Restructure Node SDK docs.

onurtemizkan opened the pull request:
Resolves: #3605

Summary:

- Removed references to client-side usage / snippets.
- Restructured the sidebar to match the flow of other JavaScript docs.
- Moved parts to `platform-includes` when possible.
- Removed outdated remarks / warnings about old versions.

AbhiPrasad commented on src/platforms/common/performance/index.mdx:
```diff
@@ -89,7 +90,7 @@ Learn more about how the options work in <PlatformLink to="/configuration/sampli
 ## Verify
 
-<PlatformSection supported={["react-native", "java.spring", "java.spring-boot", "android", "javascript", "apple", "dart", "rust"]} >
+<PlatformSection supported={["react-native", "java.spring", "java.spring-boot", "android", "javascript", "apple", "dart", "rust"]} notSupported={["node"]} >
```
Instead of making this `notSupported`, can we restructure the performance docs for Node to have the `automatic-instrumentation` and `custom-instrumentation` sections like we do in JS (https://docs.sentry.io/platforms/javascript/performance/instrumentation/) and Android (https://docs.sentry.io/platforms/android/performance/instrumentation/)

This also means we can add custom performance metric docs to Node as well following that format. We can do this in a follow-up PR though.

AbhiPrasad replied:
@onurtemizkan https://github.com/getsentry/sentry-docs/pull/5760 is blocked on this change - so let's try to include that here.

AbhiPrasad reviewed (commented):
I like this re-arrange, but will defer to @lizokm for a final ✅. We can address my comment in another PR if this gets merged in.

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_Import name: `Sentry.Integrations.InboundFilters`_
+
+This integration allows developers to ignore specific errors based on the error message or the URL from which the exception originated.
```
```suggestion
This integration allows you to ignore specific errors based on the error message or the URL from which the exception originated.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+This integration allows developers to ignore specific errors based on the error message or the URL from which the exception originated.
+
+To configure it, use the `ignoreErrors`, `denyUrls`, and `allowUrls` SDK options directly.
+(Keep in mind that `denyUrls` and `allowUrls` only work for captured exceptions, not raw message events.)
```
```suggestion
Keep in mind that `denyUrls` and `allowUrls` only work for captured exceptions, not raw message events.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/javascript.mdx:
```diff
@@ -0,0 +1,171 @@
+
+To configure this integration, use `ignoreErrors`, `denyUrls`,
+and `allowUrls` SDK options directly. Keep in mind that `denyURLs` and `allowURLs`
+work only for captured exceptions, not raw message events.
```
```suggestion only work for captured exceptions, not raw message events.
```

imatwawana replied:
```suggestion
only work for captured exceptions, not raw message events.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/javascript.mdx:
```diff
@@ -0,0 +1,171 @@
+
+_Import name: `Sentry.Integrations.FunctionToString`_
+
+This integration allows the SDK to provide original functions and method names, even when our error or breadcrumbs handlers wrap them.
```
```suggestion
This integration allows the SDK to provide original functions and method names, even when those functions or methods are wrapped by our error or breadcrumb handlers.
```

imatwawana replied:
```suggestion
This integration allows the SDK to provide original functions and method names, even when those functions or methods are wrapped by our error or breadcrumb handlers.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_Import name: `Sentry.Integrations.Console`_
+
+This integration wraps the `console` module in order to record all of its calls as breadcrumbs.
```
```suggestion
This integration wraps the `console` module to record all of its calls as breadcrumbs.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_Import name: `Sentry.Integrations.Http`_
+
+This integration wraps `http` and `https` modules to capture all network requests as breadcrumbs and/or tracing spans.
```
```suggestion
This integration wraps the `http` and `https` modules to capture all network requests as breadcrumbs and/or tracing spans.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+This integration attaches a global uncaught exception handler. It can be modified to provide a custom shutdown function.
+The `onFatalError` option is meant to perform a cleanup before the process exits, not fully prevent it from exiting.
+
+<Alert level="warning" title="Note">
```
Does this need to be a warning? Are you breaking something? If not, let's bump this down to a regular note.
```suggestion
<Alert level="warning">
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_Import name: `Sentry.Integrations.OnUnhandledRejection`_
+
+This integration attaches a global unhandled rejection handler. By default, all unhandled rejections trigger a warning and log the error. You can change this behavior using the `mode` option, which is in line with Node's CLI options: https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode.
```
Attempting to simplify the language a bit

```suggestion
This integration attaches a global unhandled rejection handler. By default, all unhandled rejections trigger a warning and log the error. You can change this behavior using the `mode` option, which is works with Node's CLI options: https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_(Available in version 6.18.0 and above)_
+
+This integration adds source file context to stack frames for captured exceptions
```
```suggestion
This integration adds source file context to stack frames for captured exceptions.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+_Import name: `Sentry.Integrations.LinkedErrors`_
+
+This integration allows you to configure linked errors. They'll be recursively read up to a specified limit and lookup will be performed by a specific key. By default, the limit is set to 5 and the key used is `"cause"`.
```
_5_ either needs to be written out or put in code font to denote it as a setting. Also, in the JS version of this there are no quotation marks on `cause`. Just highlighting that in case it's not intentional. Lastly, is this node-specific if it's also available as a default integration for JS?

```suggestion
This integration allows you to configure linked errors. They'll be recursively read up to a specified limit, and lookup will be performed by a specific key. By default, the limit is set to 5 and the key used is `"cause"`.
```

onurtemizkan replied:
- Updated `5` to use code font.
- `"cause"` should be the correct one, as it's a `string`. Updated the JS docs.
- Yes, `LinkedErrors` is not Node-specific, moved it out. 👍

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/node.mdx:
```diff
@@ -0,0 +1,184 @@
+
+## Modifying System Integrations
+
+To disable system integrations set `defaultIntegrations: false` when calling `init()`.
```
```suggestion
To disable system integrations, set `defaultIntegrations: false` when calling `init()`.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+These pluggable integrations are snippets of code that augment functionality for specific applications and/or frameworks. We document them so you can see what they do and that they can be enabled.
```
We say a lot more on the node version of this page in the intro. Not sure if you copy some of that here. Also, the last part of this sentence doesn't really make sense. Suggest:
```suggestion
These pluggable integrations are snippets of code that augment functionality for specific applications and/or frameworks. We document them so you can understand what they do and enable them, if needed.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+_Import name: `Sentry.Integrations.ExtraErrorData`_
+
+This integration extracts all non-native attributes from the Error object and attaches them to the event as the `extra` data. If the error object has a `.toJSON()` method, the `ExtraErrorData` integration will run it to extract additional information.
```
```suggestion
This integration extracts all non-native attributes from the error object and attaches them to the event as the `extra` data. If the error object has a `.toJSON()` method, the `ExtraErrorData` integration will run it to extract additional information.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+_Import name: `Sentry.Integrations.Debug`_
+
+This integration allows you to inspect the content of the processed event, that will be passed to `beforeSend` and effectively send to the Sentry SDK. It will *always* run as the last integration, no matter when it was registered.
```
Based on the Node version, I think this edit is correct
```suggestion
This integration allows you to inspect the content of the processed event that will be passed to `beforeSend`, and effectively send it to the Sentry SDK. It will *always* run as the last integration, no matter when it was registered.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+_Import name: `Sentry.Integrations.Offline`_
+
+This integration uses the web browser's [online and offline events](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events) to detect when no network connectivity is available. If offline, it saves events to the web browser's client-side storage (typically IndexedDB) then automatically uploads events when network connectivity is restored.
```
```suggestion
This integration uses the web browser's [online and offline events](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events) to detect when no network connectivity is available. If offline, it saves events to the web browser's client-side storage (typically IndexedDB), then automatically uploads events when network connectivity is restored.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+This integration uses the web browser's [online and offline events](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events) to detect when no network connectivity is available. If offline, it saves events to the web browser's client-side storage (typically IndexedDB) then automatically uploads events when network connectivity is restored.
+
+This plugin does not attempt to provide local storage or retries for other scenarios. For example, if the browser has a local area connection but no internet connection, then it may report that it's online, and Sentry's `Offline` plugin will not attempt to save or retry any send failures in this case.
```
```suggestion
This plugin does not attempt to provide local storage or retries for other scenarios. For example, if the browser has a local area connection but no internet connection, then it may report that it's online, and Sentry's `Offline` plugin will not attempt to save or retry any send failures in this case:
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+## How to Enable
+
+Install the `@sentry/integrations` package and provide a new instance with your config to `integrations` option. Include the plugin after the SDK has been loaded.
```
```suggestion
Install the `@sentry/integrations` package and provide a new instance with your config to the `integrations` option. Include the plugin after the SDK has been loaded.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+This integration allows you to apply a transformation to each frame of the stack trace. In the streamlined scenario, it can be used to change the name of the file frame it originates from, or it can be fed with an iterated function to apply any arbitrary transformation.
+
+On Windows machines, you have to use Unix paths and skip the volume letter in `root` option to enable. For example `C:\\Program Files\\Apache\\www` won’t work, however, `/Program Files/Apache/www` will.
```
```suggestion
On Windows machines, you have to use Unix paths and skip the volume letter in the `root` option to enable it. For example, `C:\\Program Files\\Apache\\www` won’t work, however, `/Program Files/Apache/www` will.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/javascript.mdx:
```diff
@@ -0,0 +1,85 @@
+
+<PlatformContent includePath="configuration/reporting-observer" />
+
+<PlatformSection supported={["node"]}>
+
+## Node specific
+
+### Transaction
+
+_Import name: `Sentry.Integrations.Transaction`_
+
+This integration tries to extract useful transaction names that will be used to distinguish the event from the rest. It walks through all stack trace frames and reads the first in-app frame's module and function name.
+
+</PlatformSection>
```
Should this be here?

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+Pluggable integrations are integrations that can be additionally enabled, to provide specific features. They're documented so you can see what they do and that they can be enabled. To enable pluggable integrations, provide a new instance with your config to the `integrations` option. For example: `integrations: [new Sentry.Integrations.Modules()]`.
```
```suggestion
Pluggable integrations are integrations that can be additionally enabled to provide specific features. They're documented so you can understand what they do and enable them, if needed. To enable pluggable integrations, provide a new instance with your config to the `integrations` option. For example: `integrations: [new Sentry.Integrations.Modules()]`.
```

imatwawana commented on src/platform-includes/configuration/integrations/default-integrations/javascript.mdx:
```diff
@@ -0,0 +1,171 @@
+
+_Import name: `Sentry.Integrations.LinkedErrors`_
+
+This integration allows you to configure linked errors. They’ll be recursively read up to a specified limit and lookup will be performed by a specific key. By default, the Sentry SDK sets the limit to five and the key used is `cause`.
```
```suggestion
This integration allows you to configure linked errors. They’ll be recursively read up to a specified limit, and lookup will be performed by a specific key. By default, the Sentry SDK sets the limit to five and the key used is `cause`.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+_Import name: `Sentry.Integrations.CaptureConsole`_
+
+This integration captures all `Console API` calls and redirects them to Sentry using `captureMessage` call.
+It then retriggers to preserve default native behavior.
```
```suggestion
It then re-triggers to preserve default native behavior.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+
+_Import name: `Sentry.Integrations.Dedupe`_
+
+This integration deduplicates certain events. It can be helpful if you are receiving many duplicate errors. Be aware that we will only compare stack traces and fingerprints.
```
```suggestion
This integration deduplicates certain events. It can be helpful if you're receiving many duplicate errors. Be aware that we will only compare stack traces and fingerprints.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+
+_Import name: `Sentry.Integrations.ExtraErrorData`_
+
+This integration extracts all non-native attributes from the Error object and attaches them to the event as the `extra` data.
```
```suggestion
This integration extracts all non-native attributes from the error object and attaches them to the event as the `extra` data.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+
+This integration allows you to apply a transformation to each frame of the stack trace. In the streamlined scenario, it can be used to change the name of the file frame it originates from, or it can be fed with an iterated function to apply any arbitrary transformation.
+
+On Windows machines, you have to use Unix paths and skip the volume letter in `root` option in order to make it work.
+For example `C:\\Program Files\\Apache\\www` won't work, however, `/Program Files/Apache/www` will.
```
```suggestion
On Windows machines, you have to use Unix paths and skip the volume letter in the `root` option to make it work.
For example, `C:\\Program Files\\Apache\\www` won't work, however, `/Program Files/Apache/www` will.
```

imatwawana commented on src/platform-includes/configuration/integrations/pluggable-integrations/node.mdx:
```diff
@@ -0,0 +1,76 @@
+
+<PlatformContent includePath="configuration/rewrite-frames" />
+
+## Node specific
```
```suggestion
## Node-specific
```

imatwawana reviewed (commented):
This probably hasn't had a good review in a long time, so I have a lot of wording edits. Most of these are trivial, but there are a few where I have questions. However, I don't understand why we've moved content that's on platform-specific pages into includes. This is increasing complexity rather than reducing it. Can someone explain?

onurtemizkan commented on src/platform-includes/configuration/breadcrumb-hints/node.mdx:
```diff
@@ -0,0 +1,8 @@
+`level` / `input`
+
+: For breadcrumbs created from console log interceptions, this holds the original console log level and the original input data to the log function.
```
I think these should be two stand-alone statements. Updating.

onurtemizkan commented on src/platform-includes/configuration/integrations/default-integrations/javascript.mdx:
````diff
@@ -0,0 +1,171 @@
+  });
+```
+
+### HttpContext (previously: UserAgent)
````
Switched to the formatting of 'New in Version ...' remarks. 👍

onurtemizkan commented:
Updated the PR

@imatwawana: 
> why we've moved content that's on platform-specific pages into includes. This is increasing complexity rather than reducing it

Yes, I was trying to reduce duplication at first, but it didn't end up as I expected. Moved content back to platform specific pages.

---

@AbhiPrasad:
> can we restructure the performance docs for Node to have the automatic-instrumentation and custom-instrumentation sections like we do in JS

Done 👍  Not sure if the steps in `automatic-instrumentation` page can be considered completely automatic, though.

imatwawana commented on src/platforms/javascript/common/configuration/integrations/plugin.mdx:
```diff
@@ -6,39 +6,43 @@ redirect_from:
 _Import name: `Sentry.Integrations.ExtraErrorData`_
 
-This integration extracts all non-native attributes from the Error object and attaches them to the event as the `extra` data. If the error object has a `.toJSON()` method, the `ExtraErrorData` integration will run it to extract additional information.
+This integration extracts all non-native attributes from the error object and attaches them to the event as the `extra` data. If the error object has a `.toJSON()` method, the `ExtraErrorData` integration will run it to extract additional information.
```
```suggestion
This integration extracts all non-native attributes from the `error` object and attaches them to the event as the `extra` data. If the error object has a `.toJSON()` method, the `ExtraErrorData` integration will run it to extract additional information.
```

imatwawana commented on src/platforms/node/common/configuration/integrations/pluggable-integrations.mdx:
```diff
@@ -51,7 +51,7 @@ Available options:
 _Import name: `Sentry.Integrations.ExtraErrorData`_
 
-This integration extracts all non-native attributes from the Error object and attaches them to the event as the `extra` data.
+This integration extracts all non-native attributes from the error object and attaches them to the event as the `extra` data.
```
```suggestion
This integration extracts all non-native attributes from the `error` object and attaches them to the event as the `extra` data.
```

imatwawana reviewed (approved):
I made a couple small formatting suggestions, but otherwise this LGTM!

mydea commented on src/platforms/node/common/configuration/integrations/default-integrations.mdx:
```diff
@@ -140,6 +125,25 @@ _Import name: `Sentry.Integrations.Modules`_
 
 This integration fetches names of all currently installed Node modules and attaches the list to the event. Once fetched, Sentry will cache the list for later reuse.
 
+## Modifying System Integrations
```
This should go below `RequestData`, I think!

onurtemizkan replied:
Yes, that makes sense! And thanks for updating!

mydea reviewed (approved):
Left one comment that should be resolved, other than this I think this is good to go!

Lms24 reviewed (commented):
LGTM! 🚀
